iavl/iavl_proof.go

```diff
@@ -17,20 +18,28 @@ type IAVLProof struct {
 	RootHash   []byte
 }
 
-func (proof *IAVLProof) Verify(key []byte, value []byte, root []byte) bool {
+func (proof *IAVLProof) Verify(key []byte, value []byte, root []byte, version int) bool {
 	if !bytes.Equal(proof.RootHash, root) {
+		//fmt.Printf("Verify Failed: Roots don't match\n")
 		return false
 	}
-	leafNode := IAVLProofLeafNode{KeyBytes: key, ValueBytes: value}
+	leafNode := IAVLProofLeafNode{KeyBytes: key, ValueBytes: value, Version: version}
 	leafHash := leafNode.Hash()
 	if !bytes.Equal(leafHash, proof.LeafHash) {
+		//fmt.Printf("Verify Failed: Leafs don't match '%X' vs '%X'\n", value, leafNode.ValueBytes)
+		//fmt.Printf("'%X'\n'%X'\n", leafHash, proof.LeafHash)
 		return false
 	}
 	hash := leafHash
 	for _, branch := range proof.InnerNodes {
 		hash = branch.Hash(hash)
 	}
-	return bytes.Equal(proof.RootHash, hash)
+	if bytes.Equal(proof.RootHash, hash) {
+		return true
+	} else {
+		//fmt.Printf("Verify Failed: Aunts don't add up\n")
```

Comment:
No, and they will get removed. It's just for debugging right now (and to remind me that I've checked certain bits of the code). 

It's scaffolding :-)